communication/data_logging.hpp

```diff
@@ -112,8 +104,10 @@ typedef struct
 	
 	uint32_t sys_id;							///< the system ID
 	
+	Console<File>* console;						///< The pointer to the console containing the file to write data to
```

Comment:
I prefer the way proposed by @basil-huber [here](https://gist.github.com/basil-huber/8832e8e977a48c3eb5fc) (seen from the outside, we pass a file to data_logging, then in its implementation data_logging uses a console to simplify things)
 
Why did you do differently?